src/main/kotlin/com/katanox/tabour/core/proto/ProtoEventPublisher.kt

```diff
@@ -9,4 +9,9 @@ abstract class ProtoEventPublisher : EventPublisher() {
         val json: String = JsonFormat.printer().print(event)
         this.publish(json, busUrl, messageGroupId)
     }
+
+    fun publishEventBatch(events: List<MessageOrBuilder>, busUrl: String, messageGroupId: String? = null) {
+        val parsedEvents = events.map { JsonFormat.printer().print(it) }
+        this.publishBatch(parsedEvents, busUrl, messageGroupId)
+    }
```

Comment:
I'm curious about this one, and am having a little trouble parsing what it's doing with all the interfaces and so on 😵 If I'm parsing the code correctly - the intention is (A) to improve performance by letting publishers publish 10 separate messages with a single sqs call, right? Or is it
(B) provide a convenience function to publish 10 separate messages (e.g. availabilityevent) as a single sqs message?

for option (A) I'd suggest we look into https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-client-side-buffering-request-batching.html which should produce a similar result, even in cases where the messages can't be batched in our client side code (e.g. 10 different hotels sending 10 different availability updates within a second).
For (B), are we sure it's a good idea to add this to tabour? I think what you're doing is (A) so it's only worth going in depth on this if I'm mistaken.

Reply:
It is mainly for (B) as we allow the users to send as many as they want and we handle the batching as an implementation detail where we use the maximum batch size and we send these messages in batchs

Reply:
Taking a look at the buffered queue it seems that it doesn't support FIFO queues, so it is not worth investing in it